docs/design/Abstract Syntax Tree.md

```diff
@@ -1,77 +1,33 @@
-# This Doc covers the ExprNode AST, which has been deprecated and removed from the Codebase. 
+# Introduction to PartiQL AST
 
-# Introduction to the ExprNode AST
+## Design concerns
 
-It seems the term "AST" is often extended to mean more things than just "Abstract Syntax Tree" and our use of the term
-is no different. A better term might have been "Abstract Semantic Tree" because our AST was defined with the goal of
-modeling the intent of the PartiQL source and *not* the exact syntax. Thus, the original SQL from which an AST was
-constituted cannot be derived, however the *semantics* of that SQL are guaranteed to be preserved. One example that
-demonstrates this is the fact that we model a `CROSS JOIN` in the same way that we model an `INNER JOIN` with a
-condition of `TRUE`. Semantically, these have the exact same function and so they also have the same representation in
-the AST.
+### Abstract Semantic Tree
+It seems the term "AST" is often extended to mean more things than just "Abstract Syntax Tree" and our use of the term is no different. A better term might have been "Abstract Semantic Tree" because our AST was defined with the goal of modeling the intent of the PartiQL source and *not* the exact syntax. Thus, the original SQL from which an AST was constituted cannot be derived, however the *semantics* of that SQL are guaranteed to be preserved. One example that demonstrates this is the fact that we model a `CROSS JOIN` in the same way that we model an `INNER JOIN` with a condition of `TRUE`. Semantically, these have the exact same functionality, so they have the same representation in the AST.
 
+### Tree Structure
 > It **is** Actually a Tree
 
 Language implementations often use the term "Abstract Syntax Tree" to refer to a data structure that is actually a
 graph. Our implementation for PartiQL's AST is a tree and *not* a graph. It contains no cycles and each node can only
 reference its children.
 
-## It's Immutable
+## Implementation
+There were multiple alternative implementations of PartiQL AST, e.g. S-expression & `ExprNode`, which were already deprecated & removed. Now we are using PIG generated AST for better maintainability. See https://github.com/partiql/partiql-ir-generator to learn more about PIG. The domain file is here, [partiql.ion](https://github.com/partiql/partiql-lang-kotlin/blob/main/lang/resources/org/partiql/type-domains/partiql.ion), where the tree structure just mirrors the PartiQL grammar. 
```

Comment:
I get the point of `the tree structure just mirrors the PartiQL grammar`, I don't think this statement is 100% correct.  I am no writing expert, but here is my attempt, feel free to take it or leave it. I would suggest to point the link to the Wiki instead of the front page though. 
```suggestion
PartiQL uses [PIG](https://github.com/partiql/partiql-ir-generator/wiki) to concisely defines the structure of our ASTs. [partiql.ion](https://github.com/partiql/partiql-lang-kotlin/blob/main/lang/resources/org/partiql/type-domains/partiql.ion) contains the type universe, which specified named constraints for every node and its attributes in the generated AST. 
```